pkg/dist/extension_descriptor.go

```diff
@@ -1,22 +1,76 @@
 package dist
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/buildpacks/lifecycle/api"
+
+	"github.com/buildpacks/pack/internal/style"
 )
 
 type ExtensionDescriptor struct {
-	WithAPI  *api.Version `toml:"api"`
-	WithInfo ModuleInfo   `toml:"extension"`
+	WithAPI          *api.Version `toml:"api"`
+	WithInfo         ModuleInfo   `toml:"extension"`
+	WithTargets      []Target     `toml:"targets,omitempty"`
+	WithWindowsBuild bool
+	WithLinuxBuild   bool
```

Comment:
I'm a bit confused by these - why are they needed?

Reply:
The WithWindowsBuild and WithLinuxBuild boolean fields in the ExtensionDescriptor struct represent flags that indicate whether the extension supports building on Windows and Linux platforms, respectively.